cpp/src/com/google/idea/blaze/cpp/BlazeConfigurationResolver.java

```diff
@@ -98,9 +100,27 @@ public BlazeConfigurationResolverResult update(
         HeaderRootTrimmer.getValidRoots(
             context, blazeProjectData, toolchainLookupMap, targetFilter, executionRootPathResolver);
     builder.setValidHeaderRoots(validHeaderRoots);
+    builder.setTargetToVersionMap(targetToVersion);
     return builder.build();
   }
 
+  @NotNull
+  private static ImmutableMap<TargetKey, String> getTargetToVersionMap(ImmutableMap<TargetKey, CToolchainIdeInfo> toolchainLookupMap, ImmutableMap<CToolchainIdeInfo, BlazeCompilerSettings> compilerSettings) {
+    Map<ExecutionRootPath, String> compilerVersionByPath =
+            compilerSettings.entrySet().stream().collect(
+                    Collectors.toMap(
+                            e -> e.getKey().getCppExecutable(),
+                            e -> e.getValue().getCompilerVersion()));
+    Map<TargetKey, String> map = toolchainLookupMap.entrySet().stream()
+            .map(e -> new AbstractMap.SimpleImmutableEntry<>(
+                    e.getKey(),
```

Comment:
- Can we replace this with `e.getKey().getLabel().toString()` to avoid going over the stream again in `BlazeCompilerInfoMapService$setState`?